core/templates/dev/head/pages/collection_player/CollectionPlayer.js

```diff
@@ -236,7 +236,8 @@ oppia.controller('CollectionPlayer', [
 
     $http.get('/collectionsummarieshandler/data', {
       params: {
-        stringified_collection_ids: JSON.stringify([$scope.collectionId])
+        stringified_collection_ids: JSON.stringify(
+          [decodeURIComponent($scope.collectionId)])
```

Comment:
Why are we applying decodeURIComponent here? $scope.collectionId should be already decoded.

In general, all encoding/decoding should happen within UrlService -- we don't want to spread that throughout the codebase.

Reply:
I meant to delete that. Will remove it.

Reply:
Ah ok. Just a tip: make sure to check the "Files Changed" tab immediately once you've pushed (or, better, do a `git diff` or `meld` before you push) to make sure that your changes are exactly what you want them to be and that you haven't left in anything spurious.

Reply:
Ok, thanks.